src/datasources/detailedholdings.datasource.js

```diff
@@ -49,35 +18,58 @@ function parseResponse(text, agencyId) {
     }
   }
 
-  const responders = obj.holdingsResponse.responder || [];
+  const responders = details.responderDetailed || [];
+  // this one goes wront ..
   for (const [key, value] of Object.entries(responders)) {
     localholdings.push({
-      localHoldingsId: (value.localHoldingsId && value.localHoldingsId.$) || "",
-      willLend: (value.willLend && value.willLend.$) || "",
-      expectedDelivery:
-        (value.expectedDelivery && value.expectedDelivery.$) || "",
+      localHoldingsId: value.holdingsItem?.[0]?.localItemId || "",
+      willLend: value.holdingsItem?.[0]?.policy || "",
+      expectedDelivery: value.holdingsItem?.[0]?.expectedDelivery || "",
     });
   }
 
   return {
-    count: responders.length,
+    count: responders?.length,
     branchId: agencyId,
     holdingstatus: localholdings,
   };
 }
 
-export async function load({ localIds, agencyId }) {
-  const soap = constructSoap(localIds, agencyId);
+/**
+ * construct the query (lookupRecord:[{bibliographicRecordId, responderId}])
+ * @param localIds
+ * @param agencyId
+ * @returns {{lookupRecord: *}}
+ */
+function construcQuery(localIds, agencyId) {
+  const localids = localIds.map((loc) => ({
+    bibliographicRecordId: loc.localIdentifier,
+    responderId: agencyId,
+  }));
+
+  return {
+    lookupRecord: localids,
+  };
+}
+
+// TODO - holdingsitems
+export async function load({ localIds, agencyId }, context) {
+  const query = construcQuery(localIds, agencyId);
 
   try {
-    const res = await request
-      .post(url)
-      .set("Content-Type", "text/xml")
-      .send(soap);
+    const response = await context.fetch(url + "detailed-holdings", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(query),
+    });
 
-    return parseResponse(res.text, agencyId);
+    return parseResponse(response?.body, agencyId);
   } catch (e) {
-    console.log(e);
+    log.error("Request to holdingsservice failed." + " Message: " + e.message);
+    console.log(e, "ERROR");
+    // @TODO what to return here
```

Comment:
console.log skal måske fjernes ? :) 
Er "// @TODO what to return here" noget der skal laves senere? :)